### PR TITLE
Simplified installation and configuration process 

### PR DESCRIPTION
**Pull request by dennis-behm**
This PR is introducing a `Setup` script that prompts the user for all settings. It guides the user to provide all configuration variables. 

The new configuration file needs to be passed to the primary and secondary DBB Git Migration Modeler commands.

Here is a sample configuration file.

```
# DBB Git Migration Modeler Configuration Settings 
# Generated at Mon Aug 19 12:59:55 CEST 2024

DBB_MODELER_HOME=/u/dbehm/git/dbb-git-migration-modeler 
DBB_MODELER_WORK=/u/dbehm/git/dbb-git-migration-modeler-mathieu 

# DBB Git Migration Modeler Working Folders 
DBB_MODELER_APPCONFIG_DIR=/u/dbehm/git/dbb-git-migration-modeler-mathieu/work-configs 
DBB_MODELER_APPLICATION_DIR=/u/dbehm/git/dbb-git-migration-modeler-mathieu/work-applications 
DBB_MODELER_LOGS=/u/dbehm/git/dbb-git-migration-modeler-mathieu/work-logs 
DBB_MODELER_METADATA_STORE_DIR=/u/dbehm/git/dbb-git-migration-modeler-mathieu/work-metadatastore 
DBB_MODELER_DEFAULT_GIT_CONFIG=/u/dbehm/git/dbb-git-migration-modeler-mathieu/git-config 

# DBB Git Migration Modeler Input files
APPLICATION_DATASETS=DBEHM.MIG.COBOL,DBEHM.MIG.COPY,DBEHM.MIG.BMS 
APPLICATION_MAPPING_FILE=/u/dbehm/git/dbb-git-migration-modeler-mathieu/applicationsMapping.yaml 
REPOSITORY_PATH_MAPPING_FILE=/u/dbehm/git/dbb-git-migration-modeler-mathieu/repositoryPathsMapping.yaml 
APPLICATION_MEMBER_TYPE_MAPPING=/u/dbehm/git/dbb-git-migration-modeler-mathieu/types.txt 
SCAN_DATASET_MEMBERS=false 
SCAN_DATASET_MEMBERS_ENCODING=IBM-1047 
TYPE_CONFIGURATIONS_FILE=/u/dbehm/git/dbb-git-migration-modeler-mathieu/typesConfigurations.yaml 
DBB_ZAPPBUILD=/var/dbb/dbb-zappbuild_300 

```

### CODE REVIEW

File: Setup.sh [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# DBB Git Migration Modeler Configuration
+
+Prolog() {
+  echo "                                                                                                            "
+  echo " DBB Git Migration Modeler                                                                                  "
+  echo "                                                                                                            "
+  echo " Script:      Setup.sh                                                                                      "
+  echo "                                                                                                            "
+  echo " Description: This script is setting up a work folder including an configuration file that is leveraged     "
+  echo "              by the DBB Git Migration Modeler proces                                                       "
```
Comment (M-DLB) [inline Setup.sh:19]:
```suggestion
  echo "              by the DBB Git Migration Modeler process.                                                     "
```

File: Setup.sh [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+#*******************************************************************************
+# DBB Git Migration Modeler Configuration
+
+Prolog() {
+  echo "                                                                                                            "
+  echo " DBB Git Migration Modeler                                                                                  "
+  echo "                                                                                                            "
+  echo " Script:      Setup.sh                                                                                      "
+  echo "                                                                                                            "
+  echo " Description: This script is setting up a work folder including an configuration file that is leveraged     "
```
Comment (M-DLB) [inline Setup.sh:18]:
```suggestion
  echo " Description: This script is setting up a work folder including an configuration file that is used          "
```

File: Setup.sh [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  echo "              by the DBB Git Migration Modeler proces                                                       "
+  echo "                                                                                                            "
+  echo "              The user will be prompted for several configuration parameters that are saved                 "
+  echo "              into a configuration file within the work folder.                                             "
+  echo "              The configuration file is passed to the Migration Modeler as a required                       "
```
Comment (M-DLB) [inline Setup.sh:23]:
```suggestion
  echo "              The configuration file is passed to the DBB Git Migration Modeler as a required                  "
```

File: Setup.sh [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+#
+Prolog
+
+# Default is the root of the Git Repo
+DBB_MODELER_HOME=$(cd "$(dirname "$0")" && pwd)
+
+# Configure DBB Migration Modeler Home
+echo "[INFO] Configure DBB Migration Modeler environment variables."
+
+echo "Please specify the DBB Migration Modeler home DBB_MODELER_HOME :"
```
Comment (M-DLB) [inline Setup.sh:38]: Could we have the prompts on one line ?
```suggestion
echo "Specify the DBB Migration Modeler Home [$DBB_MODELER_HOME] :"
```
This applies to all other prompts.

File: Setup.sh [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+echo "Leave blank to default to $DBB_MODELER_HOME"
```
Comment (M-DLB) [inline Setup.sh:39]: And remove this line :)

File: Setup.sh [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    echo "Please configure DBB Migration Modeler configuration parameter $config :"
+    echo "Leave blank to default to ${!config}"
+    read variable
+    if [ "$variable" ]; then
+        declare ${config}="${variable}"
+    fi
+done
+
+# Create work dir
+echo "[INFO] Create DBB Migration Modeler work directory $DBB_MODELER_WORK"
```
Comment (M-DLB) [inline Setup.sh:101]: Maybe we can merge the next two questions into one?
Do you want to create the '$DBB_MODELER_WORK' directory and copy samples into it?"

**Review by M-DLB — COMMENTED** [review]
Few comments :)
Thank you @dennis-behm

File: Setup.sh (hunk 5 above)
Comment (dennis-behm) [reply Setup.sh]: Done

File: Setup.sh (hunk 6 above)
Comment (dennis-behm) [reply Setup.sh]: Done

File: Setup.sh [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+            cp $DBB_MODELER_HOME/samples/*.* $DBB_MODELER_WORK/
+            rc=$?
+        fi
+    fi
+fi
+
+if [ $rc -eq 0 ]; then
+
+    # Specify input files
+    echo "[INFO] Specify DBB Migration Modeler input configuration"
```
Comment (M-DLB) [inline Setup.sh:113]:
```suggestion
    echo "[INFO] Specify DBB Git Migration Modeler input configuration"
```

File: Setup.sh [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+    echo "" >>$DBB_MODELER_WORK/DBB_GIT_MIGRATION_MODELER.config
+    echo "# DBB Git Migration Modeler Input files" >>$DBB_MODELER_WORK/DBB_GIT_MIGRATION_MODELER.config
+
+    for config in ${input_array[@]}; do
+        echo "${config}=${!config} " >>$DBB_MODELER_WORK/DBB_GIT_MIGRATION_MODELER.config
+    done
+
+    echo "Saved DBB Git Migration Modeler Configuration to $DBB_MODELER_WORK/DBB_GIT_MIGRATION_MODELER.config"
+    echo "This file will be imported by the DBB Git Migration Modeler Process"
```
Comment (M-DLB) [inline Setup.sh:147]:
```suggestion
    echo "This file will be imported by the DBB Git Migration Modeler process"
```

File: Setup.sh [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    if [ "$variable" ]; then
+        declare ${config}="${variable}"
+    fi
+done
+
+# Create work dir
+echo "[INFO] Create DBB Git Migration Modeler work directory $DBB_MODELER_WORK and copy DBB Migration Modeler configuration files to it"
+read -p "Do you want to create the directory $DBB_MODELER_WORK and copy the DBB Git Migration Modeler configuration file to it (Y/n): " variable
+
+if [[ $variable =~ ^[Yy]$ ]]; then
```
Comment (M-DLB) [inline Setup.sh:95]: I think you should also test if the variable is empty, and if it is consider it's a "Yes".

**Review by M-DLB — COMMENTED** [review]
A few comments :) Thank you @dennis-behm

File: Setup.sh (hunk 9 above)
Comment (dennis-behm) [reply Setup.sh]: @M-DLB I have addressed that with my last [commit](https://github.com/IBM/dbb-git-migration-modeler/pull/18/commits/3e194e012b30f43ba45fe081b37749868fd28dfc)

**Review by M-DLB — APPROVED** [review]
Approved, Thank you @dennis-behm
